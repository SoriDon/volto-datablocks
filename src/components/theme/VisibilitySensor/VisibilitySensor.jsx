import React from 'react';
import ReactVisibilitySensor from 'react-visibility-sensor';

const VisibilitySensor = ({
  children,
  scrollCheck = true,
  resizeCheck = true,
  partialVisibility = true,
  delayedCall = true,
  offset = { top: -50, bottom: -50 },
  useVisibilitySensor = true,
  Placeholder = () => <div>&nbsp;</div>,
  ...rest
}) => {
  const [active, setActive] = React.useState(useVisibilitySensor);

  return (
    <ReactVisibilitySensor
      scrollCheck={scrollCheck}
      resizeCheck={resizeCheck}
      partialVisibility={partialVisibility}
      delayedCall={delayedCall}
      onChange={(visible) => {
        if (visible && active) {
          setActive(false);
        }
      }}
      active={active}
      getDOMElement={(val) => {
        return val?.el;
      }}
      offset={offset}
      {...rest}
    >
      {({ isVisible }) => {
        if (isVisible || !active) {
          return children;
        }

        return <Placeholder {...rest} />;
      }}
    </ReactVisibilitySensor>
  );
};

export default VisibilitySensor;
