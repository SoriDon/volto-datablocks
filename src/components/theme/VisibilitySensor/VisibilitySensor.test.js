import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import VisibilitySensor from './VisibilitySensor';

const mockStore = configureStore();

jest.mock('@plone/volto/components', () => ({
  __esModule: true,
  UniversalLink: ({ children, href }) => {
    return <a href={href}>{children}</a>;
  },
}));

test('renders a VisibilitySensor', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
    content: {
      create: {},
    },
    connected_data_parameters: {},
  });
  const component = renderer.create(
    <Provider store={store}>
      <VisibilitySensor useVisibilitySensor={false}>
        <div>
          <p>Some content</p>
        </div>
      </VisibilitySensor>
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
