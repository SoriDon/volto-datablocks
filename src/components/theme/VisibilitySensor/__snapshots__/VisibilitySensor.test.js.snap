// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a VisibilitySensor 1`] = `
<div>
  <p>
    Some content
  </p>
</div>
`;
